service/worker/archiver/history_index_blob_test.go

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package archiver
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"github.com/uber/cadence/common/blobstore/blob"
+)
+
+type HistoryIndexBlobSuite struct {
+	*require.Assertions
+	suite.Suite
+}
+
+func TestHistoryIndexBlobSuite(t *testing.T) {
+	suite.Run(t, new(HistoryIndexBlobSuite))
+}
+
+func (s *HistoryIndexBlobSuite) SetupTest() {
+	s.Assertions = require.New(s.T())
+}
+
+func (s *HistoryIndexBlobSuite) TestNewHistoryIndexBlobKey() {
+	testCases := []struct {
+		domainID       string
+		workflowID     string
+		runID          string
+		expectError    bool
+		expectBuiltKey string
+	}{
+		{
+			domainID:    "",
+			expectError: true,
+		},
+		{
+			domainID:       "testDomainID",
+			workflowID:     "testWorkflowID",
+			runID:          "testRunID",
+			expectError:    false,
+			expectBuiltKey: "17971674567288329890367046253745284795510285995943906173973.index",
+		},
+	}
+
+	for _, tc := range testCases {
+		key, err := NewHistoryIndexBlobKey(tc.domainID, tc.workflowID, tc.runID)
```

Comment:
nit: would be nice to use sub-tests here and in other table driven tests `t.Run()`